packages/amplify-provider-awscloudformation/lib/system-config-manager.js

```diff
@@ -55,7 +62,6 @@ function setProfile(awsConfig, profileName) {
   }
 
   fs.writeFileSync(credentialsFilePath, ini.stringify(credentials));
```

Comment:
Would it be possible to change the file permission when you're writing it rather that doing it on top

Reply:
theoretically yes, but fs has a bug and I checked it's not working properly

Reply:
In what case it does not work?

Reply:
like in our case, it does not set the mode correctly

Reply:
This works: 

fs.writeFileSync(credentialsFilePath, ini.stringify(credentials), {mode: 0o600});

Updated the PR to use the above code